### PR TITLE
Knot search & intent search functionality

### LINKED ISSUE #87: Search knots
As a user, I want to search for a specific knot.

### PR DESCRIPTION
**Pull request by veikkahonkanen**
Resolves #87

### CODE REVIEW

File: src/components/intent-components/intent-list/intent-list.tsx [hunk 1]
```diff
@@ -22,6 +22,42 @@ interface Props {
...
+            <InteractiveListItem
+              icon={ <IntentIcon/> }
+              title={ intent.name ?? "" }
+              onClick={ () => onIntentClick(intent) }
+            />
+          ))
+        }
+      </List>
+    );
+  }
```
Comment (Chi-Chu319) [inline src/components/intent-components/intent-list/intent-list.tsx]: Might better to use ternary expression
```suggestion
  return intents? { 
    <List>
      {
        intents.filter(item => item.name?.toLowerCase().includes(searchValue.toLowerCase())).map(intent => (
          <InteractiveListItem
            icon={ <IntentIcon/> }
            title={ intent.name ?? "" }
            onClick={ () => onIntentClick(intent) }
          />
        ))
      }
    </List> 
  } : null

File: src/components/knot-components/knot-list/knot-list.tsx [hunk 2]
```diff
@@ -83,6 +94,32 @@ const KnotPanel: React.FC<Props> = ({ knots, onKnotClick }) => {
...
+          knots.filter(item => item.name.toLowerCase().includes(searchValue.toLowerCase())).map(knot => (
+            <InteractiveListItem
+              icon={ <KnotIcon htmlColor="#000"/> }
+              title={ knot.name }
+              onClick={ () => onKnotClick(knot) }
+            />
+          ))
+        }
+      </List>
+    );
```
Comment (Chi-Chu319) [inline src/components/knot-components/knot-list/knot-list.tsx]:
```suggestion
    return knots? { 
      <List>
        {
          knots.filter(item => item.name.toLowerCase().includes(searchValue.toLowerCase())).map(knot => (
            <InteractiveListItem
              icon={ <KnotIcon htmlColor="#000"/> }
              title={ knot.name }
              onClick={ () => onKnotClick(knot) }
            />
          ))
        }
      </List>
    } : null
```

**Review by Chi-Chu319 — CHANGES_REQUESTED** [review]
Changes requested

File: src/components/intent-components/intent-list/intent-list.tsx [hunk 3]
```diff
@@ -22,6 +22,38 @@ interface Props {
...
+          intents.filter(item => item.name?.toLowerCase().includes(searchValue.toLowerCase())).map(intent => (
+            <InteractiveListItem
+              icon={ <IntentIcon/> }
+              title={ intent.name ?? "" }
+              onClick={ () => onIntentClick(intent) }
+            />
+          ))
+        }
+      </List>
+    ) : null;
```
Comment (lauriLukkarinen) [inline src/components/intent-components/intent-list/intent-list.tsx]:
```suggestion
    if (!intents) {
      return null;
    }

    return (
      <List>
        { intents
          .filter(item => item.name?.toLowerCase().includes(searchValue.toLowerCase()))
          .map(intent =>
            <InteractiveListItem
              icon={ <IntentIcon/> }
              title={ intent.name ?? "" }
              onClick={ () => onIntentClick(intent) }
            />
          )
        }
      </List>
    );
```

File: src/components/intent-components/intent-list/intent-list.tsx [hunk 4]
```diff
@@ -52,20 +84,28 @@ const IntentPanel: React.FC<Props> = ({ intents, onIntentClick }) => {
...
+        <AccordionItem title={ strings.editorScreen.intents.defaultIntents }>
+          { renderIntentGroups(IntentType.DEFAULT) }
+        </AccordionItem>
+        <AccordionItem title={ strings.editorScreen.intents.confusedIntents }>
+          { renderIntentGroups(IntentType.CONFUSED) }
+        </AccordionItem>
+        <AccordionItem title={ strings.editorScreen.intents.redirectIntents }>
+          { renderIntentGroups(IntentType.REDIRECT) }
+        </AccordionItem>
+      </>
```
Comment (lauriLukkarinen) [inline src/components/intent-components/intent-list/intent-list.tsx]:
```suggestion
        <>
          <AccordionItem title={ strings.editorScreen.intents.normalIntents }>
            { renderIntentGroups(IntentType.NORMAL) }
          </AccordionItem>
          <AccordionItem title={ strings.editorScreen.intents.defaultIntents }>
            { renderIntentGroups(IntentType.DEFAULT) }
          </AccordionItem>
          <AccordionItem title={ strings.editorScreen.intents.confusedIntents }>
            { renderIntentGroups(IntentType.CONFUSED) }
          </AccordionItem>
          <AccordionItem title={ strings.editorScreen.intents.redirectIntents }>
            { renderIntentGroups(IntentType.REDIRECT) }
          </AccordionItem>
        </>
```

File: src/components/knot-components/knot-list/knot-list.tsx [hunk 5]
```diff
@@ -83,6 +94,28 @@ const KnotPanel: React.FC<Props> = ({ knots, onKnotClick }) => {
...
+            <InteractiveListItem
+              icon={ <KnotIcon htmlColor="#000"/> }
+              title={ knot.name }
+              onClick={ () => onKnotClick(knot) }
+            />
+          ))
+        }
+      </List>
+    ) : null;
+  }
```
Comment (lauriLukkarinen) [inline src/components/knot-components/knot-list/knot-list.tsx]:
```suggestion
  const renderSearchedKnots = () => {
    if (!knots) {
      return null;
    }

    return (
      <List>
        { knots
          .filter(item => item.name.toLowerCase().includes(searchValue.toLowerCase()))
          .map(knot =>
            <InteractiveListItem
              icon={ <KnotIcon htmlColor="#000"/> }
              title={ knot.name }
              onClick={ () => onKnotClick(knot) }
            />
          )
        }
      </List>
    );
  }
```

**Review by lauriLukkarinen — APPROVED** [review]

**Pull request merged**